### DISCUSSION TITLE
How to validate form in angular carbon is not clear

### DISCUSSION THREAD

**Discussion by mak009** [IBM/carbon-components-angular#1823] [Q&A]
I started using angular carbon component for my demo project and I also same time following the document (story) and 'carbon design system'
https://www.carbondesignsystem.com
https://angular.carbondesignsystem.com/?path=/story/components-input--label

I am sorry but in  [above link](https://www.carbondesignsystem.com) usage section is not much clear about how we can do validation of (reactive) forms and also there is no such examples with code. Every time we need to jump to story page to find-out detail about every component. This leads development process very low.

Also things like validations and error handling not mentioned there. So, I request you to provide me guidance about how we can validate form using carbon components. 

I also mailed same queried to @mattrosno he suggested me to write my doubts in this forum. So @mattrosno  @cal-smith  can you help me with validating the carbon form in angular? Thanks..!!

**Comment by cal-smith** [comment]
Hey @mak009 thanks for reaching out! How you approach validation is largely left up to you. Be sure to check out the [angular.io guide on reactive forms](https://angular.io/guide/form-validation#validating-input-in-reactive-forms). You can use the `control.errors` object to provide feedback to the user, and use [the `invalid` input](https://angular.carbondesignsystem.com/documentation/classes/src_input.label.html#invalid) to control the visibility of the invalid text.

We totally welcome documentation contributions! Definitly open a PR with clarifying instructions if you have the time!
